### PR TITLE
[scala][client] invoker package enhancement

### PR DESCRIPTION
**Pull request by sandeepjindal**
Here are the details in the issue: https://github.com/OpenAPITools/openapi-generator/issues/9378

Steps to verify the change run these commands:
-  ./mvnw clean package
- ./bin/generate-samples.sh bin/configs/scala-legacy.yaml

you should see the new invoker pacakge like this:
image

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
 @mention the@clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

### CODE REVIEW

**Review by chameleon82 — COMMENTED** [review]
Thanks for the PR @sandeepjindal.
Please ensure for most of scala clients invokerPackage by default is `mainPackage + ".core"` this PR will not override default behavior. 
Please run the script `./bin/generate-samples.sh` to regenerate all scala clients

**Comment by sandeepjindal** [comment]
> Please ensure for most of scala clients invokerPackage by default is `mainPackage + ".core"` this PR will not override default behavior.

Only below marked language framework is using invokerPackage should i check in these only:

**Comment by sandeepjindal** [comment]
@chameleon82 Thanks for the help i have pushed the fix, please help review and approve.

File: bin/configs/scala-legacy.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+generatorName: scala-httpclient-deprecated
+outputDir: samples/client/petstore/scala-httpclient-deprecated
+inputSpec: modules/openapi-generator/src/test/resources/3_0/petstore.yaml
+templateDir: modules/openapi-generator/src/main/resources/scala-httpclient
+additionalProperties:
+  apiPackage: org.openapitools.example.api
+  invokerPackage: org.openapitools.example.invoker
+  artifactId: scala-legacy-petstore
```
Comment (chameleon82) [inline bin/configs/scala-legacy.yaml:8]: @sandeepjindal  Please run and commit those files
```
./bin/generate-samples.sh bin/configs/scala-legacy.yaml
```

Comment (sandeepjindal) [reply bin/configs/scala-legacy.yaml:8]: Done please check

**Review by chameleon82 — APPROVED** [review]

**Comment by sandeepjindal** [comment]
Maintainers:
@clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)
Please help merge.

**Comment by wing328** [comment]
I ran `sbt test` on `samples/client/petstore/scala-httpclient-deprecated` but got the following errors:
```
[info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
[info]   Compilation completed in 8.364s.
[error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:40:8: ScalaJsonUtil is already defined as object ScalaJsonUtil
[error] object ScalaJsonUtil {
[error]        ^
[error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:54:7: ApiInvoker is already defined as class ApiInvoker
[error] class ApiInvoker(val mapper: ObjectMapper = ScalaJsonUtil.getJsonMapper,
[error]       ^
[error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:237:7: ApiException is already defined as class ApiException
[error] class ApiException(val code: Int, msg: String) extends RuntimeException(msg)
[error]       ^
[error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/AsyncClient.scala:9:7: AsyncClient is already defined as class AsyncClient
[error] class AsyncClient(config: SwaggerConfig) extends Closeable {
[error]       ^
[error] four errors found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 21 s, completed 12 May 2021, 10:27:43 pm
```

Does it work for you in your machine?

**Comment by sandeepjindal** [comment]
> I ran `sbt test` on `samples/client/petstore/scala-httpclient-deprecated` but got the following errors:
> 
> ```
> [info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
> [info]   Compilation completed in 8.364s.
> [error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:40:8: ScalaJsonUtil is already defined as object ScalaJsonUtil
> [error] object ScalaJsonUtil {
> [error]        ^
> [error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:54:7: ApiInvoker is already defined as class ApiInvoker
> [error] class ApiInvoker(val mapper: ObjectMapper = ScalaJsonUtil.getJsonMapper,
> [error]       ^
> [error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/ApiInvoker.scala:237:7: ApiException is already defined as class ApiException
> [error] class ApiException(val code: Int, msg: String) extends RuntimeException(msg)
> [error]       ^
> [error] /Users/williamcheng/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/src/main/scala/org/openapitools/example/invoker/AsyncClient.scala:9:7: AsyncClient is already defined as class AsyncClient
> [error] class AsyncClient(config: SwaggerConfig) extends Closeable {
> [error]       ^
> [error] four errors found
> [error] (Compile / compileIncremental) Compilation failed
> [error] Total time: 21 s, completed 12 May 2021, 10:27:43 pm
> ```
> 
> Does it work for you in your machine?

Thanks for notifying, removed duplicated files.

**Comment by sandeepjindal** [comment]
Maintainers:
@clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

please help review.
As checked the failure is not related with branch changes

**Comment by sandeepjindal** [comment]
> Maintainers:
> @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)
> 
> please help review.
> As checked the failure is not related with branch changes

May i please followup on this

**Comment by chameleon82** [comment]
@wing328 could you help to check circleci build? seems it is not pr changes related fail

**Comment by sandeepjindal** [comment]
@wing328 can i please followup on this

**Comment by wing328** [comment]
CircleCI failure already fixed in the master.

**Comment by wing328** [comment]
Tested `samples/client/petstore/scala-httpclient-deprecated` again and no longer seeing the same issue:
```
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 11 Scala sources to /home/wing328/Code/openapi-generator/samples/client/petstore/scala-httpclient-deprecated/target/scala-2.11/classes ...
[info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
[info]   Compilation completed in 32.052s.
[info] Done compiling.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/wing328/.sbt/boot/scala-2.12.7/org.scala-sbt/sbt/1.2.8/protobuf-java-3.3.1.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[info] Run completed in 143 milliseconds.
[info] Total number of tests run: 0
[info] Suites: completed 0, aborted 0
[info] Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0
[info] No tests were executed.
[success] Total time: 194 s, completed Jun 18, 2021, 3:49:24 PM
```

**Comment by sandeepjindal** [comment]
Thanks appreciated @wing328 @chameleon82

**Pull request merged**